go/tools/builders/compilepkg.go

```diff
@@ -277,20 +257,10 @@ func compileArchive(
 	// constraints.
 	compilingWithCgo := haveCgo && cgoEnabled
 
-	filterForNogo := func(slice []string) []string {
-		filtered := make([]string, 0, len(slice))
-		for _, s := range slice {
-			// Do not subject the generated empty .go file to nogo checks.
-			if s != emptyGoFilePath {
-				filtered = append(filtered, s)
-			}
-		}
-		return filtered
-	}
 	// When coverage is set, source files will be modified during instrumentation. We should only run static analysis
 	// over original source files and not the modified ones.
 	// goSrcsNogo and cgoSrcsNogo are copies of the original source files for nogo to run static analysis.
-	goSrcsNogo := filterForNogo(goSrcs)
+	goSrcsNogo := append([]string{}, goSrcs...)
```

Comment:
Nit: since 1.21, we have `slices.Clone()`.

Could you please add a comment up top to remind us about it when we raise the minimum supported Go version?